packages/core/src/agent/message-list/index.ts

```diff
@@ -373,6 +378,14 @@ ${JSON.stringify(message, null, 2)}`,
       // message3.parts: [{type: "text", text: "the weather in x is y"}]
       shouldAppendToLastAssistantMessageParts
     ) {
+      if (
+        !this.newResponseMessages.has(messageV2) &&
+        !this.newUserMessages.has(messageV2) &&
+        !this.memoryMessages.has(messageV2) &&
+        messageSource === 'user'
+      ) {
+        this.updatedResponseMessages.add(messageV2);
```

Comment:
This fixes the issue of assistant messages not being saved when used in generate calls, but we should probably have a way to differentiate them in message list so that they don't show up in the UI